core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/returnfirstnonnull/ReturnFirstNonNullLazyEvaluateUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package returnfirstnonull;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ReturnFirstNonNullLazyEvaluateUnitTest {
```

Comment:
This test does not prove that the methods are lazily evaluated, only that the end result is correct.

If you wanted to verify the lazy evaluation aspect, you would need to additionally monitor the number of times each method is called, and assert on that too.

Reply:
I have used Mockito to verify the lazy evaluation aspect. If you would prefer a different approach, please let me know